pymodbus/transport/transport.py

```diff
@@ -590,10 +590,10 @@ def close(self) -> None:
         if self.connections:
             with suppress(KeyError):
                 del self.connections[self]
-        if self.other_modem:
-            self.other_modem.other_modem = None
+        if hasattr(self, "other_modem"):
+            del self.other_modem.other_modem
```

Comment:
I thought we agreed not to delete variables.

Reply:
> This is an example of different coding styles...removing variables is a no-go to me, but in this case your code is just as good as mine...so I have no problem approving this part.

_Originally posted by @janiversen in https://github.com/pymodbus-dev/pymodbus/pull/1873#discussion_r1377223124_

You've said different things at different times (re: deleting), so I'm not sure now which you prefer.  Anyway, I can revert.

Reply:
My preference is clearly not to delete, as this is not standard python, however if it's unavoidable I can live with it,

In this case it does not seem unavoidable.